lua/null-ls/builtins/formatting/treefmt_nix.lua

```diff
@@ -0,0 +1,24 @@
+local h = require("null-ls.helpers")
+local methods = require("null-ls.methods")
+
+local FORMATTING = methods.internal.FORMATTING
+
+return h.make_builtin({
+    name = "treefmt",
+    meta = {
+        url = "https://github.com/numtide/treefmt-nix",
+        description = "Fast and convenient multi-file formatting with Nix",
+    },
+    method = FORMATTING,
+    filetypes = {},
+    generator_opts = {
+        command = "treefmt-nix",
+        args = { "--allow-missing-formatter", "--stdin", "$FILENAME" },
+        to_stdin = true,
+        -- treefmt-nix wrapper needs to be available
+        condition = function(utils)
+            return utils.is_executable("treefmt-nix")
+        end,
```

Comment:
For the record, I believe this doesn't do anything. You'd have to move it up out of `generator_opts` and pass it directly to `make_builtin`. See https://github.com/nvimtools/none-ls.nvim/pull/194 for more details.